tests/test_satfunc.py

```diff
@@ -433,6 +461,158 @@ def test_defaulted_values():
+                ]
+            ),
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KR": 0, "SATNUM": 1},
+                    {"SW": 0.1, "KR": 0.1, "SATNUM": 1},
+                    {"SW": 1, "KR": 1, "SATNUM": 1},
+                ]
+            ),
+        ),
+        (
+            # python float nan:
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KR": 0, "SATNUM": 1},
+                    {"SW": 0.1, "KR": float("nan"), "SATNUM": 1},
+                    {"SW": 1, "KR": 1, "SATNUM": 1},
+                ]
+            ),
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KR": 0, "SATNUM": 1},
+                    {"SW": 0.1, "KR": 0.1, "SATNUM": 1},
+                    {"SW": 1, "KR": 1, "SATNUM": 1},
+                ]
+            ),
+        ),
+        (
+            # It will not care about column names:
+            pd.DataFrame(
+                [
+                    {"SW": 0, "FOO": 0, "SATNUM": 1},
+                    {"SW": 0.1, "SATNUM": 1},
+                    {"SW": 1, "FOO": 1, "SATNUM": 1},
+                ]
+            ),
+            pd.DataFrame(
+                [
+                    {"SW": 0, "FOO": 0, "SATNUM": 1},
+                    {"SW": 0.1, "FOO": 0.1, "SATNUM": 1},
+                    {"SW": 1, "FOO": 1, "SATNUM": 1},
+                ]
+            ),
+        ),
+        (
+            # No extrapolation:
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KRW": 0, "SATNUM": 1},
+                    {"SW": 0.1, "KRW": np.nan, "SATNUM": 1},
+                ]
+            ),
+            pd.DataFrame(
+                [
+                    {"SW": 0, "KRW": 0, "SATNUM": 1},
+                    {"SW": 0.1, "KRW": np.nan, "SATNUM": 1},
+                ]
+            ),
+        ),
+        pytest.param(
+            pd.DataFrame([{"SW": 0}]),
+            None,
+            marks=pytest.mark.xfail(
+                raises=KeyError,
+                match="SATNUM",
+            ),
+        ),
+        pytest.param(
+            pd.DataFrame([{"SW": np.nan, "SATNUM": 1}]),
+            None,
+            marks=pytest.mark.xfail(
+                raises=ValueError, match="nan in saturation column is not allowed"
+            ),
+        ),
+        pytest.param(
+            pd.DataFrame(),
+            None,
+            marks=pytest.mark.xfail(
+                raises=AssertionError,
+                match="Could not determine a single saturation column",
+            ),
+        ),
+        pytest.param(
+            pd.DataFrame([{"FOO": 1}]),
+            None,
+            marks=pytest.mark.xfail(
+                raises=AssertionError,
+                match="Could not determine a single saturation column",
+            ),
+        ),
+        pytest.param(
+            pd.DataFrame([{"SW": 1, "SG": 1}]),
+            None,
+            marks=pytest.mark.xfail(
```

Comment:
I am a bit torn on this, I like the compactness. Personally I would put valid/invalid in two separate functions, but will leave it up to you.

Reply:
Just separate out `xfail` to e.g. `test_interpolate_defaults_xfails()` or also to rewrite the expected failures to something that "passes" pytest?

Reply:
> also to rewrite the expected failures to something that "passes" pytest?

That is my personal preference, for example:
```
@pytest.mark.parametrize(
    "error, error_msg",
    [
        (ValueError, "something"),
        (TypeError, "something"),
    ],
)
def test_my_func(error, error_msg):
    with pytest.raises(error, match=error_msg):
        raise ValueError("something")
``` 
but again, it is my personal preference, so up to you.

Reply:
I think I will skip that for now, as it would imply also rewriting other existing test functions.